src/rdkafka_topic.c

```diff
@@ -476,63 +476,81 @@ const char *rd_kafka_topic_name (const rd_kafka_topic_t *app_rkt) {
 }
 
 
-
-
-
 /**
- * @brief Update the leader for a topic+partition.
- * @returns 1 if the leader was changed, else 0, or -1 if leader is unknown.
- *
- * @locks rd_kafka_topic_wrlock(rkt) and rd_kafka_toppar_lock(rktp)
+ * @brief Update the broker for a topic+partition.
+ * 
+ * @param broker_id The id of the broker to associate the toppar with.
+ * @param rkb A reference to the broker to delegate to (must match
+ *        broker_id) or NULL if the toppar should be undelegated for
+ *        any reason.
+ * @param is_leader rd_true if broker_id is the leader, else rd_false.
+ * 
+ * @returns 1 if the broker delegation was changed, -1 if the broker
+ *          delegation was changed and is now undelegated, else 0.
+ * 
+ * @locks caller must have rd_kafka_toppar_lock(rktp)
  * @locality any
  */
-int rd_kafka_toppar_leader_update (rd_kafka_toppar_t *rktp,
-                                   int32_t leader_id, rd_kafka_broker_t *rkb) {
-
-        rktp->rktp_leader_id = leader_id;
-        if (rktp->rktp_leader_id != leader_id) {
```

Comment:
This log is still useful for the leader change, down in `if (is_leader)`

Reply:
ok, i had decided it was overly verbose, but when SHTF, additional logging here may indeed prove useful.